tests/test_python_types.cpp

```diff
@@ -575,15 +575,18 @@ test_initializer python_types([](py::module &m) {
 
     m.def("return_capsule_with_name_and_destructor_3",
         []() {
-            py::print("creating capsule");
             auto capsule=py::object(py::capsule((void *) 1234, "pointer type description",
-                [](PyObject *ptr) {py::print("destructing capsule");}));
+                [](PyObject *ptr) {
+                 if(ptr){
+                    py::print("destructing capsule");
+                 }
+            }));
             auto name=PyCapsule_GetName(capsule.ptr());
```

Comment:
We should provide enough of the interface here that we can avoid hitting the internals, particularly because the test suite also serves as a source of examples.  Let's add a simple `const char *name() const { return PyCapsule_GetName(m_ptr); }` method to get it.